src/main/java/seedu/address/logic/commands/Command.java

```diff
@@ -1,20 +1,32 @@
 package seedu.address.logic.commands;
 
+import seedu.address.logic.UndoRedoStack;
 import seedu.address.logic.commands.exceptions.CommandException;
 import seedu.address.model.Model;
 
 /**
  * Represents a command with hidden internal logic and the ability to be executed.
  */
 public abstract class Command {
+    protected Model model;
+    protected UndoRedoStack undoRedoStack;
 
     /**
      * Executes the command and returns the result message.
      *
-     * @param model {@code Model} which the command should operate on.
      * @return feedback message of the operation result for display
      * @throws CommandException If an error occurs during command execution.
      */
-    public abstract CommandResult execute(Model model) throws CommandException;
+    public abstract CommandResult execute() throws CommandException;
 
+    /**
+     * Set the required components that are required for commands to be executed
+     * If different dependencies are required, override method
+     *
+     * @param model for command to execute on
+     * @param undoRedoStack access for undoable commands
+     */
+    public void setDependencies(Model model, UndoRedoStack undoRedoStack) {
+        this.model = model;
```

Comment:
since `undoRedoStack` is not used here, maybe can overload this method instead of overriding it? I think all the overridden method is just adding `this.undoRedoStack = undoRedoStack`.

Reply:
Not quite sure how does overloading work since I need to call `setDependencies` in `LogicManager` where I don't know which type of command it is executing so I don't know which method to call. Also overriding makes it easier for us to add more parameters like `Calendar` in the future too! 😸